conda/cli/install.py

```diff
@@ -315,6 +318,61 @@ def ensure_update_specs_exist(prefix: str, specs: list[str]):
             raise PackageNotInstalledError(prefix, spec.name)
 
 
+def _assemble_environment(
+    name: str | None = None,
+    prefix: str | None = None,
+    specs: list[str] = (),
+    files: list[str] = (),
+    inject_default_packages: bool = True,  
+) -> Environment:
+    # First, let's create an 'Environment' for the information exposed in the CLI (no files)
+    if inject_default_packages:
+        names = {MatchSpec(pkg).name for pkg in specs}
+        for pkg in context.create_default_packages:
+            pkg_name = MatchSpec(pkg).name
+            if pkg_name not in names:
+                specs.append(pkg_name)
+
+    requested_packages = []
+    fetch_explicit_packages = []
+
+    for spec in specs:
+        if is_package_file(spec):
+            fetch_explicit_packages.append(spec)
+        else:
+            requested_packages.append(common.arg2spec(spec, json=context.json))
+
+    # transform explicit packages into package records
+    explicit_packages = []
+    if fetch_explicit_packages:
+        explicit_packages = get_package_records_from_explicit(fetch_explicit_packages)
+
+    cli_env = Environment(
+        name=name,
+        prefix=prefix,
+        platform=context.subdir,
+        requested_packages=requested_packages,
+        explicit_packages=explicit_packages,
+        # TODO: set the context for the environment
+        # config=context.environment_settings,
+    )
+
+    # Now let's process potential files passed via --file
+    file_envs = []
+    if files:
+        for path in files:
+            # parse the file
+            spec_hook = context.plugin_manager.get_environment_specifiers(path)
+            file_env = spec_hook.environment_spec(path).environment
+            file_envs.append(file_env)
+
+    if context.dry_run:
+        cli_env.prefix = os.path.join(mktemp(), UNUSED_ENV_NAME)
+        return Environment.merge(cli_env, *file_envs)
+    else:
+        return Environment.merge(cli_env, *file_envs)
```

Comment:
If any of the files specifies an environment name or prefix, will be overridden, or does the first one win?

Reply:
currently, the implementation in the Environment model is so that the first one wins. And, it will log if there are several prefixes specified. 
https://github.com/conda/conda/blob/main/conda/models/environment.py#L240